netutils/telnetd/telnetd_daemon.c

```diff
@@ -369,6 +389,7 @@ static int telnetd_daemon(int argc, FAR char *argv[])
  *
  ****************************************************************************/
 
+#ifndef CONFIG_NETUTILS_TELNETD_USE_POSIX_SPAWNP
 int telnetd_start(FAR struct telnetd_config_s *config)
```

Comment:
why not change telnetd to a normal application to handle both flat and kernel mode in the unified way?

Reply:
@xiaoxiang781216 
So do you mean that `telnetd_start` should create 'Telnet daemon' with `posix_spawnp` as well?

Reply:
> @xiaoxiang781216 So do you mean that `telnetd_start` should create 'Telnet daemon' with `posix_spawnp` as well? 

Yes.

> However, `telnetd_start` is just a helper function and I think this API will not be used for CONFIG_BUILD_KERNEL=y because the daemon will be created on the prompt or in the script.

We can change telnetd_start to telnetd_main and call task_create/posix_spawnp in both flat/kernel/protect build. It doesn't make sense to use the different method to launch telnetd.

Reply:
OK. I will try to modify `telnetd_start` later.

Reply:
done.